src/main/java/de/tum/in/www1/artemis/web/rest/lecture/LectureUnitResource.java

```diff
@@ -188,4 +191,19 @@ public ResponseEntity<LectureUnitForLearningPathNodeDetailsDTO> getLectureUnitFo
         authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, lectureUnit.getLecture().getCourse(), null);
         return ResponseEntity.ok(LectureUnitForLearningPathNodeDetailsDTO.of(lectureUnit));
     }
+
+    /**
+     * GET /lecture-units/:lectureUnitId : get the lecture unit with the given id.
+     *
+     * @param lectureUnitId the id of the lecture unit that should be fetched
+     * @return the ResponseEntity with status 200 (OK) and the lecture unit in the body, or with status 404 (Not Found) if the lecture unit could not be found
+     */
+    @GetMapping("lecture-units/{lectureUnitId}")
+    public ResponseEntity<LectureUnit> getLectureUnitById(@PathVariable @Valid Long lectureUnitId) {
+        log.debug("REST request to get lecture unit with id: {}", lectureUnitId);
+        var lectureUnit = lectureUnitRepository.findByIdWithCompletedUsersElseThrow(lectureUnitId);
+        lectureUnit.setCompleted(lectureUnit.isCompletedFor(userRepository.getUser()));
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, lectureUnit.getLecture().getCourse(), null);
```

Comment:
Guess you missed this comment because it was hidden by GitHub, but please also add Authorization here:
```suggestion
    @GetMapping("lecture-units/{lectureUnitId}")
    @EnforceAtLeastStudent
    public ResponseEntity<LectureUnit> getLectureUnitById(@PathVariable @Valid Long lectureUnitId) {
        log.debug("REST request to get lecture unit with id: {}", lectureUnitId);
        var lectureUnit = lectureUnitRepository.findByIdWithCompletedUsersElseThrow(lectureUnitId);
        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, lectureUnit.getLecture().getCourse(), null);
        lectureUnit.setCompleted(lectureUnit.isCompletedFor(userRepository.getUser()));
```